addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -28,19 +29,31 @@ def _default_lead_type(self):
         else:
             return 'opportunity'
 
+    def _default_country_ids(self):
+        return self.env.user.company_id.country_id
+
+    def _default_user(self):
+        return self.env.user
+
+    @api.depends('user_id')
+    def _compute_team_id(self):
+        for record in self:
+            record.team_id = record.user_id.sale_team_id
+
     name = fields.Char(string='Request Number', required=True, readonly=True, default=lambda self: _('New'), copy=False)
     state = fields.Selection([('draft', 'Draft'), ('done', 'Done'), ('error', 'Error')], string='Status', required=True, default='draft')
 
     # Request Data
-    lead_number = fields.Integer(string='Number of Leads', required=True, default=10)
+    lead_number = fields.Integer(string='Number of Leads', required=True, default=3)
     search_type = fields.Selection([('companies', 'Companies'), ('people', 'Companies and their Contacts')], string='Target', required=True, default='companies')
     error = fields.Text(string='Error', readonly=True)
 
     # Lead / Opportunity Data
+
     lead_type = fields.Selection([('lead', 'Leads'), ('opportunity', 'Opportunities')], string='Type', required=True, default=_default_lead_type)
     display_lead_label = fields.Char(compute='_compute_display_lead_label')
-    team_id = fields.Many2one('crm.team', string='Sales Team', domain="[('use_opportunities', '=', True)]")
-    user_id = fields.Many2one('res.users', string='Salesperson')
+    team_id = fields.Many2one('crm.team', string='Sales Team', domain="[('use_opportunities', '=', True)]", readonly=False, compute='_compute_team_id', store=True)
```

Comment:
You can define the field on two line for readability purpose (we do not force a characters count, but here it becomes long)